otri/validation/validators/discrepancy_validator.py

```diff
@@ -0,0 +1,105 @@
+__author__ = "Riccardo De Zen <riccardodezen98@gmail.com>"
+__version__ = "1.0"
+
+from .. import ParallelValidator
+from ..exceptions import DiscrepancyError
+
+from typing import TypeVar, Sequence, Mapping, List
+
+K = TypeVar('K')
+'''Generic atom key type.'''
+
+
+class DiscrepancyValidator(ParallelValidator):
+
+    '''
+    Validator aiming to find discrepancies between two or more Streams.
+    The Streams are checked two by two, and if their values differ by more than a certain level,
+    their atoms are marked as having a high discrepancy.
+
+    The Streams can be more than two, and the discrepancy is applied from left to right.
+    Example: given the Streams A,B and C. Maximum discrepancy level is 10%. Notice discrepancy
+    limits are inclusive.
+
+    - Check that `a * 0.9 <= b <= a * 1.1 for each a in A and b in B`.
+    - Check that `a * 0.9 <= c <= a * 1.1 for each c in A and c in C`.
+    - Check that `b * 0.9 <= c <= b * 1.1 for each b in B and c in C`.
+
+    In order to produce results that make sense, you may want to interpolate the atoms to fill in
+    order to have Streams that are parallel in the "datetime" axis. If interpolating does not suit
+    your needs, you still may want to check for gaps. See `ContinuityValidator` for that.
+    '''
+
+    def __init__(self, inputs: Sequence[str], outputs: Sequence[str], limits: Mapping[K, float]):
+        '''
+        This Validator expects the same number of Stream inputs and outputs.
+
+        Parameters:
+            inputs : str
+                Names of the inputs.
+
+            outputs : str
+                Names of the outputs.
+
+            limits : Mapping
+                Mapping of keys to their discrepancy limits, expressed as a float.
+                No checks are performed on the limits' sense, **you have been warned**.
+                Any limit is allowed, the expression is always:
+
+                for k, limit in self._limits.items():
+                    bounds = [first[k] * (1 - limit), first[k] * (1 + limit)]
+                    left = min(bounds)
+                    right = max(bounds)
+                    if not (left <= second[k] <= right):
+                        # DiscrepancyError...
+        '''
+        super().__init__(inputs, outputs)
+        self._limits = limits
+
+    def _check(self, data: List[Mapping], indexes: List[int]):
```

Comment:
Codacy found an issue: [An attribute defined in otri.validation line 383 hides this method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=6277039)